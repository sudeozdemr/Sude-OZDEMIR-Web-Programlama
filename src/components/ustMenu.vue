  <template>
  <div id="app">
    <div class="container">
      <img src="src/assets/logo.svg" />
      <input
        type="text"
        placeholder="Ürün, kategori veya marka ara"
        name="searchBar"
        id="aramaKutusu"
      />
      <a type="button" class="araBtn"
        ><FontAwesomeIcon icon="search"></FontAwesomeIcon
      ></a>
      <ul class="navBar">
        <router-link to="/uyeOl" tag="li" v-if="!currentUser"
          ><img
            src="src/assets/profil.png"
            alt="yok"
            style="width: 30px; margin-top: -10px; margin-left: 5px"
          /><a> Üye Ol </a>
        </router-link>
        <router-link to="/login" tag="li" v-if="!currentUser"
          ><img
            src="src/assets/profil.png"
            alt="yok"
            style="width: 30px; margin-top: -10px; margin-left: 5px"
          /><a> Üye Girişi </a>
        </router-link>
        <router-link
          v-if="currentUser"
          tag="li"
          to="#"
          @click.native.prevent="showInfo()"
          ><img
            src="src/assets/profil.png"
            alt="yok"
            style="width: 30px; margin-top: -10px; margin-left: 5px"
          /><a>Hesabım</a></router-link
        >
        <router-link to="/sepetim" tag="li"
          ><!-- <img
            src="src/assets/sepetikon.png"
            alt="yok"
            style="width: 30px; margin-top: -10px; margin-left: 5px"
          /> --><FontAwesomeIcon icon="shopping-cart">&#xf07a;</FontAwesomeIcon>
          <span class="badge badge-warning" id="lblCartCount">{{
            getCartCount
          }}</span
          ><a> Sepetim </a>
        </router-link>
        <!--<li><a href="http://vuex.vuejs.org/" target="_blank"><img src="src/assets/profil2.png" alt="yok" style="width:25px;margin-top:-5px;margin-left:5px">Üye Girişi</a></li>-->
        <!-- <li><a href="http://vue-loader.vuejs.org/" target="_blank"><img src="src/assets/sepetikon.png" alt="yok" style="width:30px;margin-top:-10px;margin-left:5px">Sepetim</a></li> -->
        <li>
          <a href="https://github.com/vuejs/awesome-vue" target="_blank"
            ><img
              src="src/assets/kagitucak.png"
              alt="yok"
              style="width: 30px; margin-top: -10px; margin-left: 5px"
            />Sipariş Takibi</a
          >
        </li>
        <router-link
          tag="li"
          to="#"
          class="sign-out"
          @click.native.prevent="logout()"
          v-if="currentUser"
          ><FontAwesomeIcon icon="sign-out-alt" /><a> Çıkış Yap</a></router-link
        >
      </ul>

      <ul class="Menu">
        <router-link to="/Anasayfa" tag="button"
          ><FontAwesomeIcon icon="ellipsis-v"></FontAwesomeIcon
          ><FontAwesomeIcon icon="ellipsis-v"></FontAwesomeIcon
          ><FontAwesomeIcon icon="ellipsis-v"> </FontAwesomeIcon>Tüm Lensler
        </router-link>
        <router-link to="/saydamLens" tag="li">
          <a> Saydam Lens </a>
        </router-link>
        <router-link to="/gunlukLens" tag="li">
          <a> Günlük Lens </a>
        </router-link>
        <router-link to="/renkliLens" tag="li">
          <a> Renkli Lens </a>
        </router-link>
        <router-link to="/toricLens" tag="li">
          <a> Toric Lens </a>
        </router-link>
        <router-link to="/indirimliLens" tag="li">
          <a> İndirimli Lens </a>
        </router-link>
        <router-link to="/solusyon" tag="li">
          <a> Solüsyonlar </a>
        </router-link>
        <router-link to="/gozBakim" tag="li"> <a> Göz Bakım </a> </router-link>
        <router-link to="/kampanyalar" tag="li">
          <a> Kampanyalar </a>
        </router-link>
      </ul>
    </div>
  </div>
</template>

<script>
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { library } from "@fortawesome/fontawesome-svg-core";
import { fas } from "@fortawesome/free-solid-svg-icons";
import { mapActions, mapGetters } from "vuex";
import { onAuthStateChanged } from "firebase/auth";
import { auth } from "../../firebase/config";
library.add(fas);
export default {
  name: "app",
  data() {
    return {
      msg: "Merhabalar Bu benim ilk başarılı uygulamam olabilir.",
      currentUser: null,
      //cartCount: 0,
    };
  },
  components: {
    FontAwesomeIcon,
  },
  methods: {
    ...mapActions(["logout", "fetchCart"]),

    showInfo() {
      alert(
        "Kullanıcı Bilgileriniz: \nMailiniz: " + this.currentUser.email,
        "\nID'niz: " + this.currentUser.uid
      );
    },
  },
  computed: {
   ...mapGetters([
      'getCartCount',
   ]),
  },
  created() {
    onAuthStateChanged(auth, (user) => {
      if (user) {
        this.currentUser = auth.currentUser;
      }
    });
  },
};
</script>

  <style scoped>
#app {
  font-family: "Avenir", Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  color: #2c3e50;
  margin-top: 60px;
}

h1,
h2 {
  font-weight: normal;
  border: 3px solid #2c3e50;
  background: red;
}

.navBar {
  display: inline-block;
  list-style-type: none;
  padding: 0;
  margin-left: 15px;
}

.navBar li {
  display: inline-block;
  margin: 0 5px;
}

.navBar li a {
  text-decoration: none;
  color: black;
}
.navBar li a:hover {
  color: orangered;
}
.container {
  /*width:90%;
    margin: 0 auto;*/
  text-align: center;
}
.container img {
  float: left;
  margin-left: 85px;
}
#aramaKutusu {
  display: inline-block;
  line-height: 30px;
  width: 30%;
  border: 1px solid orangered;
  border-radius: 3px;
  padding-left: 10px;
}
#aramaKutusu {
  display: inline-block;
  line-height: 30px;
  width: 30%;
  border: 1px solid orangered;

  border-radius: 3px;
}
.araBtn {
  height: 33px;
  /* border: 1px solid orangered;
    border-radius: 3px;
    border-bottom-left-radius: 0  ;
    border-top-left-radius: 0;
    background: white;*/
  margin-left: -30px;
}
.araBtn:hover {
  cursor: pointer;
}
.Menu {
  border-top: 1px solid #e3e3e3;
  border-bottom: 1px solid #e3e3e3;
  padding: 15px 0;
  box-shadow: 0px 10px 18px 10px #e3e3e3;
}
.Menu button {
  display: inline-block;
  padding: 12px 16px;
  margin: -16px 0;
  border: none;
  background: white;
  font-weight: bold;
  color: #2c3e50;
  font-size: 17px;
}
.Menu button:hover {
  background: #f60;
  color: #fff;
  border: 1px solid #f60;
  cursor: pointer;
}
.Menu li {
  border-left: 1px solid gray;
  padding-left: 15px;
  display: inline-block;
  margin: 0 5px;
}
.Menu li a {
  text-decoration: none;
  color: #2c3e50;
  font-weight: bold;
}
.Menu li a:hover {
  color: orangered;
}
.badge {
  padding-left: 9px;
  padding-right: 9px;
  -webkit-border-radius: 9px;
  -moz-border-radius: 9px;
  border-radius: 9px;
}

.label-warning[href],
.badge-warning[href] {
  background-color: #c67605;
}
#lblCartCount {
  font-size: 12px;
  background: #ff0000;
  color: #fff;
  padding: 0 3px;
  vertical-align: top;
  margin-left: -10px;
}
</style>
