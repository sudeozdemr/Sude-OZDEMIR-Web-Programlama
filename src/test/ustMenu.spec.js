import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'
import ustMenu from '../components/ustMenu.vue'

import storeConfig from "../store";
const localVue = createLocalVue();
localVue.use(Vuex);

describe('kullanıcı girişi yapıldığında', () => {

    let wrapper;
    const mockFunction = jest.fn();
    beforeEach(() => {
        wrapper = mount(ustMenu, {
            localVue,
            data() {
                return {
                    currentUser: {
                        email: 'test@test.com',
                        password: '111222',
                    }
                };
            },

            store: new Vuex.Store({
                state: { cart: [{}, {}] },
                getters: {
                    getCartCount(state) {
                        return state.cart.length;
                    }
                }
            }),
            stubs: ['router-link']
        });
    });

    it("Çıkış Yap Butonu Gözüküyor mu?", () => {
        expect(wrapper.text()).toContain("Çıkış Yap");
    });
    it("Hesabım Butonu Gözüküyor mu?", () => {
        expect(wrapper.text()).toContain("Hesabım");
    });
});
describe('Giriş yapılmadığında', () => {
    let wrapper;
    const mockFunction = jest.fn();
    beforeEach(() => {
        wrapper = mount(ustMenu, {
            localVue,
            data() {
                return {
                    currentUser: null
                };
            },
            store: new Vuex.Store({
                state: { cart: [{}, {}] },
                getters: {
                    getCartCount(state) {
                        return state.cart.length;
                    }
                }
            }),
            stubs: ['router-link']
        });
    });

    it("Üye Ol Butonu Gözüküyor mu?", () => {
        expect(wrapper.text()).toContain("Üye Ol");
    });
    it("Üye Girişi Butonu Gözüküyor mu?", () => {
        expect(wrapper.text()).toContain("Üye Girişi");
    });
    test("Snapshot Testi", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});