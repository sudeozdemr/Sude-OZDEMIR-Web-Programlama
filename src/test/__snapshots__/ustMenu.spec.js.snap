// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Giriş yapılmadığında Snapshot Testi 1`] = `
"<div id=\\"app\\">
  <div class=\\"container\\"><img src=\\"src/assets/logo.svg\\"> <input type=\\"text\\" placeholder=\\"Ürün, kategori veya marka ara\\" name=\\"searchBar\\" id=\\"aramaKutusu\\"> <a type=\\"button\\" class=\\"araBtn\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"search\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\" class=\\"svg-inline--fa fa-search fa-w-16\\">
        <path fill=\\"currentColor\\" d=\\"M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z\\" class=\\"\\"></path>
      </svg></a>
    <ul class=\\"navBar\\">
      <router-link-stub to=\\"/uyeOl\\" tag=\\"li\\"><img src=\\"src/assets/profil.png\\" alt=\\"yok\\" style=\\"width: 30px; margin-top: -10px; margin-left: 5px;\\"><a> Üye Ol </a></router-link-stub>
      <router-link-stub to=\\"/login\\" tag=\\"li\\"><img src=\\"src/assets/profil.png\\" alt=\\"yok\\" style=\\"width: 30px; margin-top: -10px; margin-left: 5px;\\"><a> Üye Girişi </a></router-link-stub>
      <!---->
      <router-link-stub to=\\"/sepetim\\" tag=\\"li\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"shopping-cart\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 576 512\\" class=\\"svg-inline--fa fa-shopping-cart fa-w-18\\">
          <path fill=\\"currentColor\\" d=\\"M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z\\" class=\\"\\"></path>
        </svg> <span id=\\"lblCartCount\\" class=\\"badge badge-warning\\">2</span><a> Sepetim </a></router-link-stub>
      <li><a href=\\"https://github.com/vuejs/awesome-vue\\" target=\\"_blank\\"><img src=\\"src/assets/kagitucak.png\\" alt=\\"yok\\" style=\\"width: 30px; margin-top: -10px; margin-left: 5px;\\">Sipariş Takibi</a></li>
      <!---->
    </ul>
    <ul class=\\"Menu\\">
      <router-link-stub to=\\"/Anasayfa\\" tag=\\"button\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"ellipsis-v\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 192 512\\" class=\\"svg-inline--fa fa-ellipsis-v fa-w-6\\">
          <path fill=\\"currentColor\\" d=\\"M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z\\" class=\\"\\"></path>
        </svg><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"ellipsis-v\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 192 512\\" class=\\"svg-inline--fa fa-ellipsis-v fa-w-6\\">
          <path fill=\\"currentColor\\" d=\\"M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z\\" class=\\"\\"></path>
        </svg><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"ellipsis-v\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 192 512\\" class=\\"svg-inline--fa fa-ellipsis-v fa-w-6\\">
          <path fill=\\"currentColor\\" d=\\"M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z\\" class=\\"\\"></path>
        </svg>Tüm Lensler
      </router-link-stub>
      <router-link-stub to=\\"/saydamLens\\" tag=\\"li\\"><a> Saydam Lens </a></router-link-stub>
      <router-link-stub to=\\"/gunlukLens\\" tag=\\"li\\"><a> Günlük Lens </a></router-link-stub>
      <router-link-stub to=\\"/renkliLens\\" tag=\\"li\\"><a> Renkli Lens </a></router-link-stub>
      <router-link-stub to=\\"/toricLens\\" tag=\\"li\\"><a> Toric Lens </a></router-link-stub>
      <router-link-stub to=\\"/indirimliLens\\" tag=\\"li\\"><a> İndirimli Lens </a></router-link-stub>
      <router-link-stub to=\\"/solusyon\\" tag=\\"li\\"><a> Solüsyonlar </a></router-link-stub>
      <router-link-stub to=\\"/gozBakim\\" tag=\\"li\\"><a> Göz Bakım </a></router-link-stub>
      <router-link-stub to=\\"/kampanyalar\\" tag=\\"li\\"><a> Kampanyalar </a></router-link-stub>
    </ul>
  </div>
</div>"
`;
